playbooks/common/openshift-cluster/upgrades/upgrade.yml

```diff
@@ -173,6 +173,18 @@
   # TODO: To better handle re-trying failed upgrades, it would be nice to check if the node
   # or docker actually needs an upgrade before proceeding. Perhaps best to save this until
   # we merge upgrade functionality into the base roles and a normal config.yml playbook run.
+  - name: Determine if node is currently scheduleable
+    command: >
+      {{ openshift.common.client_binary }} get node {{ openshift.common.hostname | lower }} -o json
+    register: node_output
+    delegate_to: "{{ groups.oo_first_master.0 }}"
+    changed_when: false
+    when: inventory_hostname in groups.oo_nodes_to_config
```

Comment:
Ok definitely do not do what I said. openshift.node.schedulable looks like it is coming from openshift_facts but *that* is not reading anything from the system, it looks like it's just assuming master = false, node = true. True state of the node is not considered so your check here is much safer/better.